great_expectations/rule_based_profiler/data_assistant/data_assistant.py

```diff
@@ -353,6 +349,70 @@ def batch_id_to_batch_identifier_display_name_map(
             for batch_id, batch in self._batches.items()
         }
 
+    def _validate_profiler_rule_name_uniqueness(self):
+        rule: Rule
+
+        profiler_rules: List[Rule] = self.profiler.rules
+        if profiler_rules is None:
+            profiler_rules = []
+
+        profiler_rule_names: Set[str] = {rule.name for rule in profiler_rules}
+
+        custom_rules: List[Rule] = self.rules
+        if custom_rules is None:
+            custom_rules = []
+
+        custom_rule_names: Set[str] = {rule.name for rule in custom_rules}
+
+        common_rule_names: Set[str] = profiler_rule_names & custom_rule_names
+        if common_rule_names:
+            raise ge_exceptions.ProfilerConfigurationError(
+                message=f"""Rule names in {self.__class__.__name__} must be unique; duplicate(s) found \
+({common_rule_names}).
+"""
+            )
```

Comment:
Thank you 🙇🏽 

Please annotate the return as `-> None`

Reply:
@cdkini Done (and added a docstring) -- thanks.